lang/cs/Org.Apache.REEF.IO/FileSystem/IFileSystem.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+// 
+//   http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace Org.Apache.REEF.IO.FileSystem
+{
+    /// <summary>
+    /// A file system abstraction.
+    /// </summary>
+    public interface IFileSystem
+    {
+        /// <summary>
+        /// Opens the given URI for reading
+        /// </summary>
+        /// <param name="fileUri"></param>
+        /// <returns></returns>
+        /// <exception cref="IOException">If the URI couldn't be opened.</exception>
+        Stream Open(Uri fileUri);
+
+        /// <summary>
+        /// Creates a new file under the given URI.
+        /// </summary>
+        /// <param name="fileUri"></param>
+        /// <returns></returns>
+        /// <exception cref="IOException">If the URI couldn't be created.</exception>
+        Stream Create(Uri fileUri);
+
+        /// <summary>
+        /// Deletes the file under the given URI.
+        /// </summary>
+        /// <param name="fileUri"></param>
+        /// <exception cref="IOException"></exception>
+        void Delete(Uri fileUri);
+
+        /// <summary>
+        /// Determines whether a file exists under the given URI.
+        /// </summary>
+        /// <param name="fileUri"></param>
+        /// <returns></returns>
+        bool Exists(Uri fileUri);
+
+        /// <summary>
+        /// Copies the file referenced  by sourceUri to destinationUri.
+        /// </summary>
+        /// <param name="sourceUri"></param>
+        /// <param name="destinationUri"></param>
+        /// <exception cref="IOException"></exception>
+        void Copy(Uri sourceUri, Uri destinationUri);
+
+        /// <summary>
+        /// Copies the remote file to a local file.
```

Comment:
Perhaps change the doc to directory if the functionality can, if not, any reason for not supporting directories?